### DISCUSSION TITLE
25 Frames PAL standard in Europe.

### DISCUSSION THREAD

**Discussion by FilipStadler** [owncast/owncast#1579] [Ideas]
I have testet owncast and its very nice - usefull but I have some problems with 25 frames - 24 frames is close but 30 frames to much compared to PAL standard so I am missing to option to pick 25 frames for output.

Is it to do a pull rtmp://somewhere/url or mayble hls ?

I use owncast in combo with Multiple Restreamer where I can pull hls and push output to owncat but also I can use
https://datarhei.github.io/ to setup streams for facebook/youtube from owncast no encoding for both platforms just push tru the streams on the rpi4.  

I have testet owncast on rpi4 with 4gb memory and on Ubuntu 21.04  Amd64 cpu with 8gb ram and with GPU Nvdia Gforce GTX 670 but even if the GPU works with my OBS I would like to maybe configure options/parameters to ffmpeg because its not working for me both input and output - In OBS I can use CBR 5500 kbps Nvidia Envenc h264 but the GTX670 is not stable on owncast or maybe my ubuntu 21.04 but maybe I need to upgrade the old GTX670 to be more Nvidia compatible ;)

**Comment by gabek** [comment]
Hi there. The next release will have 25fps as an option.  See this discussion: https://github.com/owncast/owncast/issues/1138

As for the other things, I'm having a hard time completely understanding what you're asking.  I don't know what you mean by push and pull of rtmp and hls.  Also that link you posted seems to be broken.

It sounds like you want to be able to manually customize the transcoding process?  That's unlikely something we'd be adding, as focusing on making things easier and working for everyone is the goal, so having a text box where you can type your own incantations goes in the wrong direction there. But if there are actual issues with your setup then that's something that we should fix.  Please let me know what "not stable on owncast" specifically means.  Of course you could always build from source if you wanted to change what's going on internally.

**Reply by FilipStadler** [reply]
Happy for the 25 frames in future ;)

The tool I use with owncast is this https://datarhei.github.io/restreamer/docs/why.html  so I can push rtmp to owncast with maybe HLS input.
Also I can use it to do streaming to youtube and facebook at the same time.

Anyway on my raspberry pi with a elgato usb3 capture card I can stream 1080p/25 frames using SRT this really give a Teradek Vidiu like solution where I can make a hotspot on my phone and transmit to my OBS using udp so or I can use ffmpeg to get srt and make rtmp to the owncast like this:

#
while true; do
ffmpeg -i "srt://:1935?mode=listener&passphrase=keys123456" -c copy -f flv rtmp://127.0.0.1:1935/live/keys
done

srt-tools is part of Raspian Bullseye - its just another way to get input to owncast - its very useful and fast protocol compared to rtmp.

Ok sorry I maybe speaking about alot of stuff on the same time - the need for tweks was because my GTX670 is not working stable with owncast even if it works fine with kdenlive and other  that use nvidia gpu.
